src/handlers/swap_quote_handlers.ts

```diff
@@ -0,0 +1,198 @@
+        };
+        this._swapQuoter = new SwapQuoter(this._provider, orderbook, swapQuoterOpts);
+        this._swapQuoteConsumer = new SwapQuoteConsumer(this._provider);
+        this._web3Wrapper = new Web3Wrapper(this._provider);
+    }
+    public async calculateSwapQuoteAsync(
+        buyTokenAddress: string,
+        sellTokenAddress: string,
+        buyAmount: BigNumber | undefined,
+        sellAmount: BigNumber | undefined,
+        takerAddress: string | undefined,
+        isETHSell: boolean = false,
+        slippagePercentage?: number,
+        providedGasPrice?: BigNumber,
+    ): Promise<SwapQuoteResponse> {
+        let swapQuote;
+        if (sellAmount !== undefined) {
+            swapQuote = await this._swapQuoter.getMarketSellSwapQuoteAsync(
+                buyTokenAddress,
+                sellTokenAddress,
+                sellAmount,
+                { slippagePercentage, gasPrice: providedGasPrice },
+            );
+        } else if (buyAmount !== undefined) {
+            swapQuote = await this._swapQuoter.getMarketBuySwapQuoteAsync(
+                buyTokenAddress,
+                sellTokenAddress,
+                buyAmount,
+                { slippagePercentage, gasPrice: providedGasPrice },
+            );
+        } else {
+            throw new Error('sellAmount or buyAmount required');
+        }
+        const {
+            makerAssetAmount,
+            takerAssetAmount,
+            protocolFeeInWeiAmount: protocolFee,
+        } = swapQuote.worstCaseQuoteInfo;
+        const { orders, gasPrice } = swapQuote;
+
+        // If ETH was requested and the operation is Buy we should use the Forwarder
+        const extensionContractType =
+            isETHSell && swapQuote.type === MarketOperation.Buy
+                ? ExtensionContractType.Forwarder
+                : ExtensionContractType.None;
+        const {
+            calldataHexString: data,
+            ethAmount: value,
+            toAddress: to,
+        } = await this._swapQuoteConsumer.getCalldataOrThrowAsync(swapQuote, {
+            useExtensionContract: extensionContractType,
+        });
+
+        let gas = ZERO;
+        if (takerAddress) {
+            gas = await this._estimiateGasOrThrowRevertErrorAsync({
+                to,
+                data,
+                from: takerAddress,
+                value,
+                gasPrice,
+            });
+        }
+
+        const apiSwapQuote: SwapQuoteResponse = {
+            // TODO not everything has 18 decimals
+            price: makerAssetAmount.dividedBy(takerAssetAmount).decimalPlaces(DEFAULT_TOKEN_DECIMALS),
+            to,
+            data,
+            value,
+            gasPrice,
+            protocolFee,
+            gas,
+            makerAssetAmount,
+            takerAssetAmount,
+            orders,
+        };
+        return apiSwapQuote;
+    }
+    public async getSwapQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
```

Comment:
Should this be private?

Reply:
This one is public as it used for the express/HTTP/request. 

`calculateSwapQuoteAsync` is public for using in upcomming tests without having to munge express request/response.

Reply:
oops yes i meant `calculateSwapQuoteAsync`